src/fprime_gds/common/fpy/compiler.py

```diff
@@ -0,0 +1,1709 @@
+import traceback
+
+from fprime_gds.common.fpy.bytecode.directives import (
+    serialize_directives,
+    BINARY_COMPARISON_DIRECTIVES,
+    FLOAT_INEQUALITY_DIRECTIVES,
+    INT_EQUALITY_DIRECTIVES,
+    MAX_SERIALIZABLE_REGISTER_SIZE,
+    INT_SIGNED_INEQUALITY_DIRECTIVES,
+    INT_UNSIGNED_INEQUALITY_DIRECTIVES,
+    AndDirective,
+    CmdDirective,
+    DeserSerReg1Directive,
+    DeserSerReg2Directive,
+    DeserSerReg4Directive,
+    DeserSerReg8Directive,
+    Directive,
+    FloatExtendDirective,
+    IntEqualDirective,
+    ExitDirective,
+    FloatEqualDirective,
+    FloatNotEqualDirective,
+    GetPrmDirective,
+    GetTlmDirective,
+    GotoDirective,
+    IfDirective,
+    NotDirective,
+    IntNotEqualDirective,
+    OrDirective,
+    SetSerRegDirective,
+    SetRegDirective,
+    SignedIntToFloatDirective,
+    UnsignedIntToFloatDirective,
+    WaitAbsDirective,
+    WaitRelDirective,
+)
+from fprime_gds.common.loaders.ch_json_loader import ChJsonLoader
+from fprime_gds.common.loaders.cmd_json_loader import CmdJsonLoader
+from fprime_gds.common.loaders.prm_json_loader import PrmJsonLoader
+from fprime_gds.common.templates.ch_template import ChTemplate
+from fprime_gds.common.templates.cmd_template import CmdTemplate
+from fprime_gds.common.templates.prm_template import PrmTemplate
+from fprime.common.models.serialize.time_type import TimeType
+from fprime.common.models.serialize.enum_type import EnumType
+from fprime.common.models.serialize.serializable_type import SerializableType
+from fprime.common.models.serialize.array_type import ArrayType
+from fprime.common.models.serialize.numerical_types import (
+    U32Type,
+    U16Type,
+    U64Type,
+    U8Type,
+    I16Type,
+    I32Type,
+    I64Type,
+    I8Type,
+    F32Type,
+    F64Type,
+    FloatType,
+    IntegerType,
+    NumericalType,
+)
+from fprime.common.models.serialize.string_type import StringType
+from fprime.common.models.serialize.bool_type import BoolType
+from fprime_gds.common.fpy.parser import (
+    AstAnd,
+    AstBoolean,
+    AstComparison,
+    AstElif,
+    AstElifs,
+    AstExpr,
+    AstGetAttr,
+    AstGetItem,
+    AstNot,
+    AstNumber,
+    AstOr,
+    AstReference,
+    AstString,
+    Ast,
+    AstTest,
+    AstBody,
+    AstLiteral,
+    AstIf,
+    AstAssign,
+    AstFuncCall,
+    AstVar,
+    parse,
+)
+from fprime.common.models.serialize.type_base import BaseType as FppType
+
+NUMERIC_TYPES = (
+    U32Type,
+    U16Type,
+    U64Type,
+    U8Type,
+    I16Type,
+    I32Type,
+    I64Type,
+    I8Type,
+    F32Type,
+    F64Type,
+)
+INTEGER_TYPES = (
+    U32Type,
+    U16Type,
+    U64Type,
+    U8Type,
+    I16Type,
+    I32Type,
+    I64Type,
+    I8Type,
+)
+SIGNED_INTEGER_TYPES = (
+    I16Type,
+    I32Type,
+    I64Type,
+    I8Type,
+)
+UNSIGNED_INTEGER_TYPES = (
+    U32Type,
+    U16Type,
+    U64Type,
+    U8Type,
+)
+FLOAT_TYPES = (
+    F32Type,
+    F64Type,
+)
+
+
+FppTypeClass = type[FppType]
```

Comment:
Please document what line 135 is intended to mean. It sounds like the intended meaning of the RHS is something like "for all types T such that T is a subtype of FppType, type[T]." I'm not sure what it actually means in Python, or if it even has a defined meaning, since Python type annotations aren't checked by default.